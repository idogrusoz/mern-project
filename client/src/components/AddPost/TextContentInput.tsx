import { TextField } from "@material-ui/core";
import React, { Dispatch, FunctionComponent, SetStateAction } from "react";

type TextContentInputProps = {
    textContent: string;
    setTextContent: Dispatch<SetStateAction<string>>;
};

const TextContentInput: FunctionComponent<TextContentInputProps> = ({ textContent, setTextContent }) => {
    return (
        <TextField
            id="standard-basic"
            label="Text"
            type="text"
            onChange={(e: { target: { value: any } }) => setTextContent(e.target.value)}
            value={textContent}
            fullWidth
        />
    );
};

export default TextContentInput;
