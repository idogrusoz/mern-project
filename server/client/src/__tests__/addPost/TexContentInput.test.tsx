import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import TextContentInput from "../../components/AddPost/TextContentInput";

describe("TextContentInput test", () => {
    const props = {
        textContent: "",
        setTextContent: jest.fn(() => {}),
    };

    const wrapper = shallow(<TextContentInput {...props} />);
    it("renders without crash", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
