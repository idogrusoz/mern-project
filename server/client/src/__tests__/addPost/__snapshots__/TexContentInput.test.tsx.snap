// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextContentInput test renders without crash 1`] = `
<WithStyles(ForwardRef(TextField))
  fullWidth={true}
  id="standard-basic"
  label="Text"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
